scripts/Diffraction/isis_powder/hrpd.py

```diff
@@ -60,6 +62,48 @@ def mask_prompt_pulses_if_necessary(self, ws_list):
         for ws in ws_list:
             self._mask_prompt_pulses(ws)
 
+    def should_subtract_empty_inst(self):
+        return self._inst_settings.subtract_empty_inst
+
+    def create_solid_angle_corrections(self, vanadium, run_details):
+        """
+        Creates the solid angle corrections from a vanadium run, only applicable on HRPD otherwise return None
+        :param vanadium: The vanadium used to create this
+        :param run_details: the run details of to use
+        """
+        if not self._inst_settings.do_solid_angle:
+            return None
+        solid_angle = mantid.SolidAngle(InputWorkspace=vanadium)
+        scale = mantid.CreateSingleValuedWorkspace(DataValue='100')
+        correction = mantid.Multiply(LHSWorkspace=solid_angle, RHSWorkspace=scale)
+        eff = mantid.Divide(LHSWorkspace=vanadium, RHSWorkspace=correction)
+        eff = mantid.ConvertUnits(InputWorkspace=eff, Target='Wavelength')
+        eff = mantid.Integration(InputWorkspace=eff, RangeLower='1.3999999999999999', RangeUpper='3')
+        correction = mantid.Multiply(LHSWorkspace=correction, RHSWorkspace=eff)
+        scale = mantid.CreateSingleValuedWorkspace(DataValue='100000')
+        correction = mantid.Divide(LHSWorkspace=correction, RHSWorkspace=scale)
+        name = common.generate_splined_name(run_details.run_number,[])
+        path = run_details.van_paths
+        name = "sac"+name
+        mantid.SaveNexus(InputWorkspace=correction,Filename=os.path.join(path, name))
+        common.remove_intermediate_workspace(solid_angle)
+        common.remove_intermediate_workspace(scale)
+        common.remove_intermediate_workspace(eff)
+        common.remove_intermediate_workspace(correction)
+
+    def get_solid_angle_corrections(self, vanadium, run_details):
+        if not self._inst_settings.do_solid_angle:
+            return None
+        name = common.generate_splined_name(vanadium, [])
+        path = run_details.van_paths
+        name = "sac" + name
```

Comment:
```suggestion
        name = "sac" + common.generate_splined_name(vanadium, [])
```
To avoid creating a variable only to change it one line later.
If this is an issue of clarity for the script users, I would swap lines 98 and 99, so `name` changes are consecutive